WalletWasabi.Fluent/Behaviors/HideShowBehavior.cs

```diff
@@ -0,0 +1,65 @@
+using System;
+using System.ComponentModel;
+using System.Reactive.Disposables;
+using System.Reactive.Linq;
+using System.Runtime.InteropServices;
+using Avalonia.Controls;
+using ReactiveUI;
+using WalletWasabi.Services;
+
+namespace WalletWasabi.Fluent.Behaviors
+{
+	public class HideShowBehavior : DisposingBehavior<Window>
+	{
+		protected override void OnAttached(CompositeDisposable disposables)
+		{
+			if (AssociatedObject is null)
+			{
+				return;
+			}
+
+			// On macOs the Hide/Show is a natural feature.
+			if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+			{
+				Observable
+					.FromEventPattern(Services.SingleInstanceChecker, nameof(SingleInstanceChecker.OtherInstanceStarted))
+					.ObserveOn(RxApp.MainThreadScheduler)
+					.Subscribe(_ =>
+					{
+						if (AssociatedObject.WindowState == WindowState.Minimized)
+						{
+							AssociatedObject.WindowState = (WindowState)Enum.Parse(typeof(WindowState), Services.UiConfig.WindowState);
+						}
+
+						// Fixes: https://github.com/zkSNACKs/WalletWasabi/issues/6309
```

Comment:
New Avalonia release is on its way and the fix for this is already merged. https://github.com/AvaloniaUI/Avalonia/pull/6555
Consider do not merge this quick fix, since this is surely not a critical bug. No need to double touch this part.
It is up to you, I approve because the code LGTM.